src/us/kbase/common/utils/CondorUtils.java

```diff
@@ -100,8 +110,46 @@ public static CondorResponse runProcess(String[] condorCommand) throws IOExcepti
         return new CondorResponse(stdOutMessage, stdErrMessage);
     }
 
+
+    /**
+     * Parse out requirements frm client groups
+     *
+     * @param clientGroupsAndRequirements
+     * @return a map of client_groups, resource_requirements, and classAds
+     */
+    public static HashMap<String, String> clientGroupsAndRequirements(String clientGroupsAndRequirements) {
+
+        String[] items = clientGroupsAndRequirements.split(",");
+        String clientGroup = cleanCondorInputs(items[0]);
+        HashMap<String, String> reqs = new HashMap<String, String>();
+
+        reqs.put("client_group", clientGroup);
+
+        List<String> requirementsStatement = new ArrayList<String>();
+        requirementsStatement.add(String.format("(CLIENTGROUP == \"%s\")", clientGroup));
+
+
+        for (int i = 1; i < items.length; i++) {
+            String condorInput = cleanCondorInputs(items[i]);
+            if (condorInput.contains("=")) {
+                String[] keyValue = condorInput.split("=");
+                if (special_cases.contains(keyValue[0])) {
+                    reqs.put(keyValue[0], keyValue[1]);
+                } else {
+                    requirementsStatement.add(String.format("(%s == \"%s\")", keyValue[0], keyValue[1]));
+                }
+            } else {
+                requirementsStatement.add(String.format("(%s)", condorInput));
+            }
+        }
+        reqs.put("requirements_statement", "(" + String.join(" && ", requirementsStatement) + ")");
```

Comment:
It would be useful for debugging to log what the requirements are.

Reply:
So the requirements are dumped to the submission script file. Is that what you mean?

Reply:
@sychan